src/cpp/rtps/xmlparser/XMLProfileManager.cpp

```diff
@@ -171,15 +171,28 @@ void XMLProfileManager::loadDefaultXMLFile()
     {
         loadXMLFile(file_path);
     }
+
+    char skip_xml[1];
+    size = 1;
+    error_t ret = getenv_s(&size, skip_xml, size, "SKIP_DEFAULT_XML_FILE");
+
+    if (ret == 0 && skip_xml[0] == "1")
+    {
+        loadXMLFile(DEFAULT_FASTRTPS_PROFILES);
+    }
 #else
+
     if (const char* file_path = std::getenv(DEFAULT_FASTRTPS_ENV_VARIABLE))
     {
         loadXMLFile(file_path);
     }
-#endif // ifdef _WIN32
 
-    // Try to load the default XML file.
-    loadXMLFile(DEFAULT_FASTRTPS_PROFILES);
+    // Try to load the default XML file unless set to false
+    if (!std::getenv("SKIP_DEFAULT_XML_FILE"))
```

Comment:
On `WIN32` you are checking whether the value of the variable is `1`, however there is no check in Linux. I'd add the same for consistency